x-pack/plugins/apm/server/lib/helpers/create_es_client/call_client_with_debug.ts

```diff
@@ -41,27 +42,23 @@ export async function callClientWithDebug({
   }
 
   if (debug) {
-    const highlightColor = esError ? 'bgRed' : 'inverse';
     const diff = process.hrtime(startTime);
-    const duration = `${Math.round(diff[0] * 1000 + diff[1] / 1e6)}ms`;
-    const routeInfo = `${request.route.method.toUpperCase()} ${
-      request.route.path
-    }`;
+    const duration = Math.round(diff[0] * 1000 + diff[1] / 1e6);
 
-    console.log(
-      chalk.bold[highlightColor](`=== Debug: ${routeInfo} (${duration}) ===`)
-    );
+    outputToConsole({ operationName, params, request, esError, duration });
 
-    if (operationName === 'search') {
-      console.log(`GET ${params.index}/_${operationName}`);
-      console.log(formatObj(params.body));
-    } else {
-      console.log(chalk.bold('ES operation:'), operationName);
+    // TODO: add check to verify that the user is superuser or equivalent
+    const isSuperuser = true;
+    if (isSuperuser) {
+      request._debugQueries = request._debugQueries ?? [];
 
-      console.log(chalk.bold('ES query:'));
-      console.log(formatObj(params));
+      request._debugQueries.push({
```

Comment:
Maybe `setupRequest` can return a `respond` method, and we wrap responses as: `respond({ hasData: true })`. Then we don't have to modify the request object, we'd need to pass some kind of tracking function to create_apm_event_client and call_client_with_debug.

Reply:
I'm not following. How would I access the `response` object from within `callClientWithDebug`?
`response` would have to be supplied to each route, which in turn would have to pass it on to `setupRequest` like:

```ts
const setup = await setupRequest(context, request, response);
```